torchdynamo/guards.py

```diff
@@ -153,8 +184,10 @@ def arg_ref(self, guard: Guard):
 
     def TYPE_MATCH(self, guard: Guard):
         # ___check_type_id is same as `id(type(x)) == y`
-        self.code.append(
-            f"___check_type_id({self.arg_ref(guard)}, {self.id_ref(type(self.get(guard.name)))})"
+        self.append_code_for_check_type(
```

Comment:
I like the aspect of your PR that refactors how guards are accumulated such that there is an opportunity to organize them and potentially use this to also help the error messages we print.